lib/widgets/screens/explore/explore.dart

```diff
@@ -1,57 +1,92 @@
 import 'package:flutter/material.dart';
 import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:mm_flutter_app/__generated/schema/operations_user.graphql.dart';
 import 'package:mm_flutter_app/constants/app_constants.dart';
+import 'package:mm_flutter_app/providers/user_provider.dart';
+import 'package:mm_flutter_app/providers/base/operation_result.dart';
+import 'package:mm_flutter_app/providers/models/explore_card_filters_model.dart';
+import 'package:mm_flutter_app/utilities/utility.dart';
 import 'package:mm_flutter_app/widgets/atoms/explore_filter.dart';
 import 'package:mm_flutter_app/widgets/molecules/profile_quick_view_card.dart';
+import 'package:provider/provider.dart';
 
 import '../../../utilities/navigation_mixin.dart';
 
 class ExploreCardScroll extends StatefulWidget {
-  const ExploreCardScroll({super.key});
+  final UserProvider userProvider;
+  final ExploreCardFiltersModel exploreCardFilters;
+
+  const ExploreCardScroll({
+    super.key,
+    required this.userProvider,
+    required this.exploreCardFilters,
+  });
 
   @override
   State<ExploreCardScroll> createState() => _ExploreCardScrollState();
 }
 
 class _ExploreCardScrollState extends State<ExploreCardScroll> {
-  final int requestSize = 5;
-  List<bool> isSelected = [];
-  List<ProfileQuickViewInfo> cardInfo = [];
+  static const int profileIncrement = 5;
+
+  int _profiles = 0;
+  Future<OperationResult<Mutation$CreateUserSearch$createUserSearch>>?
+      _recommendedUsers;
 
   void _loadMoreRecommendations() {
-    // future todo:  load the actual backend request into cardInfo here instead
-    for (int i = 0; i < requestSize; i++) {
-      isSelected.add(false);
-      if (cardInfo.isEmpty) {
-        cardInfo.add(createRecommendedMentorExample());
-      } else if (cardInfo.length == 1) {
-        cardInfo.add(createRecommendedEntrepreneurExample());
-      } else {
-        cardInfo.add(createRegularMentorExample());
-      }
-    }
+    _profiles += profileIncrement;
+    _recommendedUsers = widget.userProvider.createUserSearch(
```

Comment:
This is a call that immediately returns a response, but the results aren't yet generated by the backend when it returns. A successful response to this mutation can be regarded as a sign that the backend is performing a search. `CreateUserSearch` will only return an error if the input to the search is somehow malformed or if the user has insufficient permissions.

We need to poll for `findUserSearchById` and check the `UserSearch.runInfos.finishedAt` to know if the search is complete. On that API call, we can use the `UserSearch`'s `topFoundUsers` to get the first 10 users.

After we know that a search is complete, for the paginated search results, the `findUserSearchResults` query can be used.

Reply:
Just now I checked and verified the searches are actually finding results on the remote server, so that shouldn't be an issue.

Reply:
I've not tested this with the real backend. This issue looks fair. I've for now added another `FutureBuilder` that fetches results. There is no polling yet, though.